.github/workflows/build.yml

```diff
@@ -12,10 +12,12 @@ jobs:
 
       - uses: actions/cache@v4
         with:
-          key: build-and-test-${{ hashFiles('**/pom.xml') }}
+          key: maven-test-${{ hashFiles('**/pom.xml') }}
           path: ~/.m2/repository
           restore-keys: |
-            ${{ runner.os }}-maven-
+            maven-test-${{ hashFiles('**/pom.xml') }}
```

Comment:
I don't think you need a restore key that matches the key exactly. You always get a cache hit on a direct match.